### PR TITLE
ObjUtil adds support for concatenating id lists

### PR DESCRIPTION
**Pull request by karlek**
These convenience functions will protect against typos when concatenating id lists for spells or available buildings, etc. 

Usage:
```go
setStructuresBuilt(asList(ID_GUARD_TOWER, ID_CANNON_TOWER))
setNormalAbilities(asList(ID_FIREBALL, ID_FIRE_NOVA))
```

Also add extension functions for:

- string.fourchar2int() returns int
- int.int2fourchar() returns string

### CODE REVIEW

File: wurst/objediting/ObjUtil.wurst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package ObjUtil
+
+import ObjectIds
+import Assets
+import LinkedList
+
+function asList(LinkedList<int> idList) returns string
```
Comment (Frotty) [inline wurst/objediting/ObjUtil.wurst]: I'd prefer a generic solution that could be repurposed like so https://bin.wurstlang.org/wunomidoxe.wurst

File: wurst/objediting/ObjectIds.wurst [hunk 2]
```diff
@@ -31,7 +31,12 @@ public function fourchar2int(string value) returns int
 	
 	return result	
 	
+public function string.fourchar2int() returns int
```
[the remark this answers is not in the document]
Comment (karlek) [reply wurst/objediting/ObjectIds.wurst]: Haha yeah.

Btw, I really dislike the name `fourchar2int` and `int2fourchar`. We have renamed them in our map to `id2s` and `s2id` :cactus:

**Comment by Frotty** [comment]
the linkedlists created from the asListGeneric calls still aren't destroyed in the asList functions, right?
I think the joinBy function should be merged with the one in `StringUtils`
Other than that looks good

**Comment by karlek** [comment]
Suggestions added in latest commit. Is there a more clean way to get the result from a method on the list and then destroy the list? @Frotty

**Comment by Cokemonkey11** [comment]
> Is there a more clean way to get the result from a method on the list and then destroy the list?

I don't think so. Wurst doesn't have RAII or anything like that.

Looks good to me :+1:

**Comment by Frotty** [comment]
varargs are now implemented. Please update this PR and resolve conflicts. 👍

**Comment by karlek** [comment]
Will do! I'll be back from my travels this weekend 👍

**Comment by karlek** [comment]
@Frotty. Ready for launch :rocket:

**Comment by karlek** [comment]
It creates a cyclic import with `toRawCode()` and `fromRawCode()`, so those functions would need to move to `ObjUtil`. But, the importer wouldn't see any difference with `import public ObjUtil` in `ObjectIds`. 

What do you think about that? It would remove the small import.

**Comment by Frotty** [comment]
sure, if ObjectIds is public import then

**Comment by Frotty** [comment]
perhaps objUtil should just be a part of ObjectIds, since it's so small now.
Then public import ObjectIds for the pleasure.

**Comment by karlek** [comment]
I agree with the last comment, I'll merge them and make a new commit.

**Comment by Frotty** [comment]
Sweet, finally merged 🍻

**Pull request merged**